vmngclient/api/repository_api.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import os
+from enum import Enum
+from typing import Dict, List, Union
+
+from attr import define, field
+
+from vmngclient.dataclasses import Device
+from vmngclient.session import Session
+from vmngclient.utils.creation_tools import FIELD_NAME, create_dataclass, get_logger_name
+
+logger = logging.getLogger(get_logger_name(__name__))
+
+
+class DeviceCategory(Enum):
+    VSMART = "controllers"
+    VBOND = "controllers"
+    VEDGE = "vedges"
+    CEDGE = "vedges"
+    VMANAGE = "controllers"
+
+
+@define
+class DeviceSoftwareRepository:
+    installed_versions: List[str] = field(default=None)
+    available_versions: List[str] = field(default=None, metadata={FIELD_NAME: "availableVersions"})
+    current_version: str = field(default=None, metadata={FIELD_NAME: "version"})
+    default_version: str = field(default=None, metadata={FIELD_NAME: "defaultVersion"})
+    device_id: str = field(default=None, metadata={FIELD_NAME: "uuid"})
+
+
+class RepositoryAPI:
```

Comment:
This class usage is unclear. It is used both as API namespace and container for data.
I don't think we need to store anything other than session in API class

Reply:
Fixed